course/api/views.py

```diff
@@ -64,6 +95,112 @@ def get_queryset(self):
     def get_object(self):
         return self.get_member_object('instance', 'Course')
 
+    def set_teachers(self, course: CourseInstance, data: dict) -> None:
+        teachers = []
+        for i in data['teachers']:
+            userprofile: UserProfile = None
+            try:
+                user = User.objects.get(username=i)
+            except User.DoesNotExist:
+                # If user does not exist, create a new user.
+                # If external authentication (e.g. Shibboleth) is used, other
+                # attributes will be updated when user logs in for the first time.
+                user = User.objects.create_user(i)
+
+            userprofile = UserProfile.objects.get(user=user)
+            teachers.append(userprofile)
+
+        course.set_teachers(teachers)
+
+    def create(self, request: HttpRequest, version) -> Response:
+        if (not request.user.is_superuser):
+            return Response(
+                'Only admins can add new course instance.',
+                status=status.HTTP_403_FORBIDDEN
+            )
+
+        data = request.data
+        form = CourseInstanceForm(data)
+
+        if not form.is_valid():
+            return Response({'status': 'error', 'errors': form.errors}, status=status.HTTP_400_BAD_REQUEST)
+
+        else:
+            code = form.cleaned_data.get('code')
+            course: Course = Course.objects.filter(code=code).first()
+            if not course:
+                try:
+                    course = Course.objects.create(
+                        name = form.cleaned_data.get('name'),
```

Comment:
The docs could be more explicit about `name` and `course_url`. They are only used if a new Course is created for the course code, otherwise they are ignored. Existing Course records are never modified in this method, which I suppose is the way we should do this now. The API docs could simply be more explicit.